### PR TITLE
feat(functions): folder migration

### PR DESCRIPTION
**Pull request by JamesP-KC**
## Describe your changes

Azure Function and API endpoint to perform creation of Folders for cases which are being migrated.

The differences between Folder setup for migrated cases vs those created manually via CBOS are:
1. There are additional Folders for migrated cases (e.g. `Correspondence` folders within stage folders, and other additional sub-folders)
2. The root level `Correspondence` Folder is not present for migrated cases
3. Additional `Archived documentation` Folder to put any miscellaneous files during migration that are not mappable into the new Folder structure

The Function takes a list of case references, and sends a request to the API for each. The API creates the Folders for the case.

The document migration function will map legacy folders to these new ones. Any that don't match will fall back to being organised by the document's `stage`, or otherwise be added to the `Archived documentation` folder

## Issue ticket number and link

BOAS-1403

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/api/src/server/migration/migrators/folder-migrator.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * @typedef {import('pins-data-model').Schemas.Folder} FolderModel
+ * @typedef {import('apps/api/src/database/schema.d.ts').Folder} Folder
+ */
+
+import { getCaseIdFromRef } from './utils.js';
+import * as folderRepository from '#repositories/folder.repository.js';
+import { defaultCaseFoldersForMigration } from './folder/folder.js';
+
+/**
+ * @param caseReference
```
Comment (HadrianHughesKC) [inline apps/api/src/server/migration/migrators/folder-migrator.js:11]: Looks like it should be
```suggestion
 * @param {string} caseReference
```

File: apps/api/src/server/repositories/folder.repository.js [hunk 2]
```diff
@@ -223,15 +223,16 @@ const mapFolderTemplateWithCaseId = (caseId, folder) => {
  * Returns an array of promises
  *
  * @param {number} caseId
+ * @param folders
```
Comment (HadrianHughesKC) [inline apps/api/src/server/repositories/folder.repository.js:226]: Should this be
```suggestion
 * @param {Folder[]} folders
```

or something similar?

**Review by francescobrenci-AR — DISMISSED** [review]

**Review by BenSurgisonPINS — APPROVED** [review]
lgtm

File: apps/api/src/server/repositories/folder.repository.js (hunk 2 above)
Comment (JamesP-KC) [reply apps/api/src/server/repositories/folder.repository.js]: I'll use `FolderTemplate` - that is the type already assigned to `defaultCaseFolders` which is the default value for that arg

**Review by HadrianHughesKC — APPROVED** [review]

**Pull request merged**